pkg/asset/machines/azure/machines.go

```diff
@@ -49,12 +50,19 @@ func Machines(clusterID string, config *types.InstallConfig, pool *types.Machine
 	}
 	var machines []machineapi.Machine
 	machineSetProvider := &machineapi.AzureMachineProviderSpec{}
+
+	networkResourceGroup, virtualNetworkName, subnets, err := getNetworkInfo(platform, clusterID, role, 0)
```

Comment:
Argument list for getNetworkInfo() has changed to :
`func getNetworkInfo(platform *azure.Platform, clusterID, role string, subnetZones int) (string, string, []string, error) `
Not sure why it is hardcoded to 0 here. 

Also, there is no need to change the signature of `Provider()`. We can easily call getNetworkInfo() from there too.  No additional parameters are needed if you continue calling getNetworkInfo() from within Provider() with subnetZones hardcoded to 0.